src/main/java/com/aol/cyclops/types/IterableFunctor.java

```diff
@@ -44,21 +46,34 @@ A potentially asynchronous merge operation where data from each publisher may ar
      * 
      * 
      */
-    default  ReactiveSeq<T> mergePublisher(Collection<? extends Publisher<T>> publishers, QueueFactory<T> factory){
+    default  ReactiveSeq<T> mergePublisher(Iterable<? extends Publisher<T>> publishers, QueueFactory<T> factory){
         Counter c = new Counter();
-        c.active.set(publishers.size()+1);
-        QueueBasedSubscriber<T> init = QueueBasedSubscriber.subscriber(factory,c,publishers.size());
+        
+        int size = 0;
+        Iterator<? extends Publisher<T>> it = publishers.iterator();
```

Comment:
I think it won't :)

If we call

      mergePublisher(Seq.of(pub1,pub2,pub3), factory);

It will blow up because at line 55-59 we consumer the Iterable (and Seq is Traverse once), then at line 68 we attempt to consume it again (which will give the Stream has been operated on already error).